02.async-await/AsyncAwait.Task2.CodeReviewChallenge/Middleware/StatisticMiddleware.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AsyncAwait.Task2.CodeReviewChallenge.Headers;
+using CloudServices.Interfaces;
+using Microsoft.AspNetCore.Http;
+
+namespace AsyncAwait.Task2.CodeReviewChallenge.Middleware
+{
+    public class StatisticMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        private readonly IStatisticService _statisticService;
+
+        public StatisticMiddleware(RequestDelegate next, IStatisticService statisticService)
+        {
+            _next = next;
+            _statisticService = statisticService ?? throw new ArgumentNullException(nameof(statisticService));
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {   
+            string path = context.Request.Path;
+
+            Task staticRegTask = Task.Run(
+                () => _statisticService.RegisterVisitAsync(path)
```

Comment:
Replace creating of `staticRegTask` task by `await _statisticService.RegisterVisitAsync`. No need to wrap it and make overhead on worker threads